clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/ArtifactController.java

```diff
@@ -67,19 +69,24 @@ StreamingResponseBody fetch(@RequestBody Artifact artifact) {
 
   @RequestMapping(method = RequestMethod.GET, value = "/account/{accountName}/names")
   List<String> getNames(
-      @PathVariable("accountName") String accountName, @RequestParam(value = "type") String type) {
+      @PathVariable("accountName") String accountName,
+      @RequestParam(value = "type") String artifactType) {
     ArtifactCredentials credentials =
-        artifactCredentialsRepository.getCredentialsForType(accountName, type);
+        artifactCredentialsRepository.getCredentialsForType(accountName, artifactType);
     return credentials.getArtifactNames();
   }
 
   @RequestMapping(method = RequestMethod.GET, value = "/account/{accountName}/versions")
   List<String> getVersions(
       @PathVariable("accountName") String accountName,
-      @RequestParam(value = "type") String type,
+      @RequestParam(value = "type") String artifactType,
       @RequestParam(value = "artifactName") String artifactName) {
     ArtifactCredentials credentials =
-        artifactCredentialsRepository.getCredentialsForType(accountName, type);
+        artifactCredentialsRepository.getCredentialsForType(accountName, artifactType);
     return credentials.getArtifactVersions(artifactName);
   }
+
+  @ExceptionHandler(MissingCredentialsException.class)
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  public void handleMissingCredentials() {}
```

Comment:
Do you figure this belongs in kork eventually?  Like [here](https://github.com/spinnaker/kork/blob/3214d577916fef40de909db7cc9da2ed828d63a3/kork-web/src/main/java/com/netflix/spinnaker/kork/web/exceptions/GenericExceptionHandlers.java)?

Reply:
Yes but it's a regression that needs to be backported so no kork.

Reply:
Aaah, ok.  Makes sense.